### ISSUE TITLE
Redirect from getStaticProps only redirects once

### ISSUE THREAD

**Issue by balazsorban44** [vercel/next.js#18735]
# Bug report

## Describe the bug
When I go to a page that should be redirected with `getStaticProps` redirect, it is redirected.

Going to the page that should be redirected again, I get some props that look to be internal, and also get an error in the terminal.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. clone https://github.com/balazsorban44/next-example
2. build with `yarn build` and run in production mode with `yarn start`
3. Go to http://localhost:3000/should-redirect
4. You will be redirected to http://localhost:3000/redirected
5. Go to http://localhost:3000/should-redirect again
6. You will stay on http://localhost:3000/should-redirect, and some potentially internal props are shown
7. The terminal also shows an `Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client` error. (see scrennshot for more details)

## Expected behavior

Whenever I go to a page that returns a redirect property, I expect the page to redirect to `destination`

## Screenshots

A screenshot of what is shown the second time I go to an url that should redirect. The first time, the redirect happened, and I am presented with `/redirected` both in the URL and in the content.
Note: I am using a `pre` tag to show all the `props` sent to the page component
![image](https://user-images.githubusercontent.com/18369201/97981101-16ee8480-1dd2-11eb-83a1-bf916cf3bba9.png)

```sh
Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at ServerResponse.setHeader (_http_outgoing.js:526:11)
    at handleRedirect (/home/balazs/projects/redirect-gsp/node_modules/next/dist/next-server/server/render.js:17:1734)
    at renderToHTML (/home/balazs/projects/redirect-gsp/node_modules/next/dist/next-server/server/render.js:28:1094)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async /home/balazs/projects/redirect-gsp/node_modules/next/dist/next-server/server/next-server.js:100:66
    at async __wrapper (/home/balazs/projects/redirect-gsp/node_modules/next/dist/lib/coalesced-function.js:1:330)
    at async Server.renderToHTMLWithComponents (/home/balazs/projects/redirect-gsp/node_modules/next/dist/next-server/server/next-server.js:125:376)
    at async Server.renderToHTML (/home/balazs/projects/redirect-gsp/node_modules/next/dist/next-server/server/next-server.js:126:711)
    at async Server.render (/home/balazs/projects/redirect-gsp/node_modules/next/dist/next-server/server/next-server.js:69:236)
    at async Object.fn (/home/balazs/projects/redirect-gsp/node_modules/next/dist/next-server/server/next-server.js:54:264) {
  code: 'ERR_HTTP_HEADERS_SENT'
}
```

## System information

- OS: Ubuntu 20.04
- Browser: Chrome 88.0.4300.0 (Official Build) dev (64-bit) *AND* Firefox 82.0 (64-bit)
- Version of Next.js: 10.0.0
- Version of Node.js: v12.19.0

## Additional context

I have used `create-next-app` for this example, but it occurs in my other Next.js projects that I currently upgraded to next@10 as well. (Meaning it is not `create-next-app` specific)

Question in addition. How is trailing slash handled in these cases? If I go to `http://localhost:3000/should-redirect/` would this have any different effect? As far as I understand, pages with a trailing slash are automatically redirected to their non-trailing slash counterpart. Would this mess something up here?

**Comment by ijjk** [comment]
Hi, I'm having trouble reproducing this with the provided reproduction. Visiting http://localhost:3000/should-redirect always redirects with the latest version of Next.js. Is there another step that could be causing this, is there a stale cache in your browser possibly?

**Comment by balazsorban44** [comment]
@ijjk I am using incognito mode. I'll test with different browsers.

**Comment by ijjk** [comment]
@balazsorban44 I see, thanks for the additional information, I'm able to reproduce now with production mode

**Comment by balazsorban44** [comment]
Forgot to mention it was only in production mode, my bad!

**Comment by balazsorban44** [comment]
Confirmed to be fixed in `10.0.2-canary.0`, thanks!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
